src/main/java/net/dv8tion/jda/api/interactions/commands/build/OptionData.java

```diff
@@ -448,6 +519,8 @@ public DataObject toData()
                     .map(entry -> DataObject.empty().put("name", entry.getKey()).put("value", entry.getValue()))
                     .collect(Collectors.toList())));
         }
+        if (type == OptionType.CHANNEL && !channelTypes.isEmpty())
```

Comment:
@MinnDevelopment I think the empty check should be removed here because an empty array would basically make the option unrestricted if it previously was restricted

Reply:
Sending an empty array makes no sense, and it's not clear if the current behavior is even intended.

Reply:
@MinnDevelopment seems like it's intended https://github.com/discord/discord-api-docs/pull/3825#issuecomment-922479196

Reply:
No.

Reply:
it's literally how it's supposed to be so why not? it's how the api works

Reply:
> Sending an empty array makes no sense

It is optional and defaults to all channels. There is no reason to send an empty array.

Reply:
ah, I get what you're saying now. I was thinking of modifying an option to unrestrict it again but that doesn't really make sense

Reply:
Yeah, you always completely replace options, so it will just be the default again.